core/src/main/java/io/aiven/klaw/config/MigrationUtility.java

```diff
@@ -40,9 +40,11 @@ public class MigrationUtility implements InitializingBean {
 
   @Autowired private ApplicationContext context;
 
-  @Override
-  public void afterPropertiesSet() throws Exception {
-
+  @SchedulerLock(
+      name = "TaskScheduler_MigrationUtility",
+      lockAtLeastFor = "PT10M",
+      lockAtMostFor = "PT60M")
+  public void startMigration() throws Exception {
```

Comment:
Same here we should make these @Value if we can.

Reply:
I aslo think we can make a slight improvement on the migration utility as it has a time stamp in the db for when a migration for a version is complete, we can probably look at this and make sure it only executes once by the multi instance deployment so they dont all try and do it one after the other.